### PR TITLE
Support "!!binary" syntax (decodes base64 string)

### PR DESCRIPTION
**Pull request by johnnytemp**
Only supported in folded-scalar (lines ending with | or >) form, or inline form.

Example Yaml:
## items

name: !!binary "SGVsbG8sIHdvcmxkLg=="
desc: !!binary |
SGVsbG8sIH
dvcmxkLg==

### CODE REVIEW

**Comment by fabpot** [comment]
ping @xabbuh

File: src/Symfony/Component/Yaml/Inline.php [hunk 1]
```diff
@@ -21,6 +21,7 @@
  */
 class Inline
 {
+    const REGEX_TAG_PATTERN = '((?P<tag>![\w!.\/:-]+) +)?';
```
Comment (xabbuh) [inline src/Symfony/Component/Yaml/Inline.php]: This should be moved to the `Parser` class.

File: src/Symfony/Component/Yaml/Tests/InlineTest.php [hunk 2]
```diff
@@ -304,6 +304,9 @@ public function getTestsForParseWithMapObjects()
             array('-12', -12),
             array('"quoted string"', 'quoted string'),
             array("'quoted string'", 'quoted string'),
+			array('!!binary "SGVsbG8sIHdvcmxkLg=="', 'Hello, world.'),
+            array('!!float 1230', 1230.0),
+			array('!!unknown "string"', '!!unknown "string"'),
```
Comment (xabbuh) [inline src/Symfony/Component/Yaml/Tests/InlineTest.php]: indentation is broken here

File: src/Symfony/Component/Yaml/Parser.php [hunk 3]
```diff
@@ -488,10 +488,14 @@ private function parseValue($value, $exceptionOnInvalidType, $objectSupport, $ob
             return $this->refs[$value];
         }
 
-        if (preg_match('/^'.self::BLOCK_SCALAR_HEADER_PATTERN.'$/', $value, $matches)) {
+        if (preg_match('/^'.Inline::REGEX_TAG_PATTERN.self::BLOCK_SCALAR_HEADER_PATTERN.'$/', $value, $matches)) {
             $modifiers = isset($matches['modifiers']) ? $matches['modifiers'] : '';
 
-            return $this->parseBlockScalar($matches['separator'], preg_replace('#\d+#', '', $modifiers), (int) abs($modifiers));
+            $output = $this->parseBlockScalar($matches['separator'], preg_replace('#\d+#', '', $modifiers), (int) abs($modifiers));
+            if (isset($matches['tag']) && $matches['tag'] == '!!binary') {
```
Comment (xabbuh) [inline src/Symfony/Component/Yaml/Parser.php]: Please use strict comparison (`===`) here.

File: src/Symfony/Component/Yaml/Tests/ParserTest.php [hunk 4]
```diff
@@ -788,6 +788,18 @@ public function testNestedFoldedStringBlockWithComments()
         ));
     }
 
+    public function testFoldedBinaryBlock()
```
Comment (xabbuh) [inline src/Symfony/Component/Yaml/Tests/ParserTest.php]: `testBinaryTag` sounds more reasonable to me and I would pass the different YAML strings through a data provider.

File: src/Symfony/Component/Yaml/Parser.php (hunk 3 above)
Comment (paradajozsef) [inline src/Symfony/Component/Yaml/Parser.php]: + this condition should be in Yoda stlye.

Comment (xabbuh) [inline src/Symfony/Component/Yaml/Parser.php]: indeed

**Comment by xabbuh** [comment]
Status: Needs work

**Comment by stof** [comment]
What about dumping ? Should we detect binary strings and dump them this way too ?

**Comment by johnnytemp** [comment]
Hi all, feel free to edit and merge the changes on your own name.  Thank you.  :-)

**Comment by fabpot** [comment]
@xabbuh Can you take over this PR?

**Comment by xabbuh** [comment]
Will do.

**Comment by fabpot** [comment]
Thanks, closing this one then as there is a related linked issue.

**Pull request closed without merging**